### PR TITLE
[Feature] Paf detector

### PR DESCRIPTION
**Pull request by luminxu**
1. Refactor bottom-up inference (disentangle heatmaps and tags).
2. Add 'align_corners' in config.
3. Add paf detector & head & loss & post-processing & demo

### CODE REVIEW

File: mmpose/core/post_processing/group.py [hunk 1]
```diff
@@ -404,3 +404,272 @@ def parse(self, heatmaps, tags, adjust=True, refine=True):
             ans = [ans]
 
         return ans, scores
+
+
+class PAFParser:
```
Comment (jin-s13) [inline mmpose/core/post_processing/group.py:409]: rm this

File: mmpose/core/evaluation/__init__.py [hunk 2]
```diff
@@ -8,9 +9,20 @@
                             pose_pck_accuracy, post_dark_udp)
 
 __all__ = [
-    'EvalHook', 'DistEvalHook', 'pose_pck_accuracy', 'keypoints_from_heatmaps',
-    'keypoints_from_regression', 'keypoint_pck_accuracy', 'keypoint_auc',
-    'keypoint_epe', 'get_group_preds', 'get_multi_stage_outputs',
-    'aggregate_results', 'compute_similarity_transform', 'post_dark_udp',
-    'keypoint_mpjpe'
+    'EvalHook',
```
Comment (jin-s13) [inline mmpose/core/evaluation/__init__.py:12]: It looks strange.

File: mmpose/models/keypoint_heads/paf_head.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
+import torch.nn as nn
+
+from mmpose.models.builder import build_head
+from ..registry import HEADS
+
+
+@HEADS.register_module()
+class PAFHead(nn.Module):
+    """Bottom-up PAF head.
```
Comment (jin-s13) [inline mmpose/models/keypoint_heads/paf_head.py:9]: add paper ref.

**Comment by jin-s13** [comment]
Unittest fails.

File: mmpose/models/keypoint_heads/bottom_up_higher_resolution_head.py [hunk 4]
```diff
@@ -180,18 +180,18 @@ def get_loss(self, output, targets, masks, joints):
...
-            output (torch.Tensor[NxKxHxW]): Output heatmaps.
-            targets(List(torch.Tensor[NxKxHxW])): Multi-scale target heatmaps.
-            masks(List(torch.Tensor[NxHxW])): Masks of multi-scale target
-                                              heatmaps
-            joints(List(torch.Tensor[NxMxKx2])): Joints of multi-scale target
-                                                 heatmaps for ae loss
+            outputs (List(torch.Tensor[NxKxHxW])): Multi-scale output heatmaps.
+            targets (List(torch.Tensor[NxKxHxW])): Multi-scale target heatmaps.
+            masks (List(torch.Tensor[NxHxW])): Masks of multi-scale target
+                                               heatmaps
```
Comment (jin-s13) [inline mmpose/models/keypoint_heads/bottom_up_higher_resolution_head.py:186]: Use a `Tab` to indent.

File: mmpose/models/detectors/paf.py [hunk 5]
```diff
@@ -0,0 +1,415 @@
...
+from mmpose.core.evaluation import (aggregate_results_paf, get_group_preds,
+                                    get_multi_stage_outputs_paf)
+from mmpose.core.post_processing.group import PAFParser
+from .. import builder
+from ..registry import POSENETS
+from .base import BasePose
+
+
+@POSENETS.register_module()
+class BottomUp(BasePose):
```
Comment (jin-s13) [inline mmpose/models/detectors/paf.py:20]: Rename it.

**Comment by jin-s13** [comment]
Need more careful check before merge.
